### PR TITLE
Added docs for ONNX 1.17 covering logging, tracing, and QNN EP Profiling

### PR DESCRIPTION
**Pull request by ivberg**
### Description
Added docs for ONNX 1.17 covering logging, tracing, and QNN EP Profiling

### Motivation and Context
- ONNX Logging has not been documented
- ONNX Tracing with Windows has barely been documented
- ONNX 1.17 has new tracing and QNN EP Profiling

PRs: #16259,  #18201, #18882, #19397

### CODE REVIEW

**Review by HectorSVC — APPROVED** [review]
:shipit:

File: docs/performance/tune-performance/profiling-tools.md [hunk 1]
```diff
@@ -38,6 +38,34 @@ In both cases, you will get a JSON file which contains the detailed performance
...
+
+- Provider Name: Microsoft.ML.ONNXRuntime  
+- Provider GUID: 3a26b1ff-7484-7484-7484-15261f42614d  
+- Keywords: Profiling = 0x100  per [logging.h](https://github.com/ivberg/onnxruntime/blob/user/ivberg/ETWRundown/include/onnxruntime/core/common/logging/logging.h#L81)  
+- Level: 
+  - 5 (VERBOSE) = profiling_level=basic (good details without perf loss)
+  - greater than 5 = profiling_level=detailed (individual ops are logged with inference perf hit)  
+- Event: [QNNProfilingEvent](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc#L1083)
+
+## CUDA Profiling
```
Comment (HectorSVC) [inline docs/performance/tune-performance/profiling-tools.md:67]: >CUDA [](start = 3, length = 4)

change to GPU since it has ROCm also

Comment (ivberg) [reply docs/performance/tune-performance/profiling-tools.md]: done

**Review by HectorSVC — APPROVED** [review]
:shipit:

**Pull request merged**